### PR TITLE
feat(gtest): add support of reply code to `Log`

### PR DESCRIPTION
**Pull request by shamilsan**
- Added `Log::auto_reply_builder` to create an empty `Log` entry for auto-reply.
- Added the `Log::reply_code` function to explicitly set the reply code in the `Log` entry.

@gear-tech/dev 

## Release Notes

- **Added a possibility to check auto-replies in `gtest`.**
- Now, it is possible to create `gtest::Log` entries with a custom reply code to check whether we have received an auto-reply or not.

### CODE REVIEW

**Review by clearloop — APPROVED** [review]

File: gtest/src/log.rs [hunk 1]
```diff
@@ -144,6 +144,14 @@ impl Log {
         log
     }
 
+    /// Create a `Log` with a reply code of `SuccessReplyReason::Auto`.
+    pub fn auto_reply_builder() -> Self {
```
Comment (breathx) [inline gtest/src/log.rs:148]: auto replies never has payload, better to somehow reflect it

**Review by breathx — CHANGES_REQUESTED** [review]

File: gtest/src/log.rs (hunk 1 above)
Comment (shamilsan) [reply gtest/src/log.rs:148]: > auto replies never has payload, better to somehow reflect it

How? Add a separate `AutoReplyLog` type? Seems to be overkill.

Comment (NikVolf) [reply gtest/src/log.rs:148]: Why is it overkill?

Comment (breathx) [reply gtest/src/log.rs:148]: At least panic once added payload for such kind of Log

Comment (shamilsan) [reply gtest/src/log.rs:148]: > At least panic once added payload for such kind of Log

Sounds reasonable; I added it.

**Review by breathx — CHANGES_REQUESTED** [review]

File: gtest/src/log.rs [hunk 2]
```diff
@@ -153,6 +161,10 @@ impl Log {
             panic!("Payload was already set for this log");
         }
 
+        if let Some(ReplyCode::Success(SuccessReplyReason::Auto)) = self.reply_code {
```
[the remark this answers is not in the document]
Comment (shamilsan) [reply gtest/src/log.rs:165]: Added `#[track_caller]` to ell functions with `panic!` and `expect`.

File: gtest/src/log.rs [hunk 3]
```diff
@@ -176,6 +188,17 @@ impl Log {
...
         self
     }
+
+    /// Set the reply code for this log.
+    pub fn reply_code(mut self, reply_code: ReplyCode) -> Self {
+        if self.reply_code.is_some() {
+            panic!("Reply code was already set for this log");
+        }
+
+        self.reply_code = Some(reply_code);
```
[the remark this answers is not in the document]
Comment (shamilsan) [reply gtest/src/log.rs:205]: Good catch, thank you! Added a check and panic.

**Review by breathx — APPROVED** [review]

**Pull request merged**